### PR TITLE
net-irc/scrollz: version bump, fix build for clang 16 and lto.

### PR DESCRIPTION
**Pull request by Schievel1**
Closes: https://bugs.gentoo.org/731206
    Closes: https://bugs.gentoo.org/861467
    Closes: https://bugs.gentoo.org/870907

### CODE REVIEW

File: net-irc/scrollz/scrollz-2.3.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	gmp? ( dev-libs/gmp:0= )
+	ssl? (
+		gnutls? ( net-libs/gnutls:0= )
+		!gnutls? ( dev-libs/openssl:0= )
+	)
+"
+RDEPEND="${DEPEND}"
+
+PATCHES=(
+	"${FILESDIR}/${P}-fix-clang16-build.patch"
```
Comment (Flowdalic) [inline net-irc/scrollz/scrollz-2.3.1.ebuild:34]: The fix-clang16-build.patch is very large, consider obtaining it via github, e.g. 

```
SRC_URI="
    https://www.scrollz.info/download/${MY_P}.tar.gz
    https://github.com/ScrollZ/ScrollZ/pull/30/commits/344bfdc1e3df6be47971c318dd5492a66e688cd2.patch -> scrollz-2.3.1-fix-clang-and-lto-build.patch
"

PATCHES=(
    # https://github.com/ScrollZ/ScrollZ/pull/30
    "${DESTDIR}/scrollz-2.3.1-fix-clang-and-lto-build.patch"
)

Comment (Schievel1) [reply net-irc/scrollz/scrollz-2.3.1.ebuild:34]: Sam usually hosts those when he prams the PR

**Comment by Schievel1** [comment]
Will Squash commits when done with this

File: net-irc/scrollz/scrollz-2.3.1.ebuild (hunk 1 above)
Comment (Flowdalic) [reply net-irc/scrollz/scrollz-2.3.1.ebuild:35]: Sure, but if you want to spare him some work, then you could simply use the github link. No strong opinion though, just wanted to point out the possibility.

Comment (Schievel1) [reply net-irc/scrollz/scrollz-2.3.1.ebuild:35]: problem is that this patch on the PR on GitHub goes on the HEAD of their master branch, while this here goes on 2.3.1.
There are already a few rejects for that patch from here when I patched their HEAD.
I will correct those on the PR and then this patch won't works on 2.3.1 here anymore.

Comment (Flowdalic) [reply net-irc/scrollz/scrollz-2.3.1.ebuild:35]: You could create a branch "2.3.1-clang16-and-lto-fixes" containing the patch which applies on 2.3.1 on your github fork on the project and then use this as base for the SRC_URI. I assume that would be the idiomatic thing to do.

**Comment by thesamesam** [comment]
```
blowfish.c:132:8: error: type defaults to ‘int’ in declaration of ‘FishDecrypt’ [-Werror=implicit-int]
  132 | extern FishDecrypt _((char *, char *, char *, int));
      |        ^~~~~~~~~~~
blowfish.c:133:8: error: type defaults to ‘int’ in declaration of ‘FishEncrypt’ [-Werror=implicit-int]
  133 | extern FishEncrypt _((char *, char *, char *, int, int));
      |        ^~~~~~~~~~~
```

Full log: http://sprunge.us/G4LAtj

**Comment by Schievel1** [comment]
> ```
> blowfish.c:132:8: error: type defaults to ‘int’ in declaration of ‘FishDecrypt’ [-Werror=implicit-int]
>   132 | extern FishDecrypt _((char *, char *, char *, int));
>       |        ^~~~~~~~~~~
> blowfish.c:133:8: error: type defaults to ‘int’ in declaration of ‘FishEncrypt’ [-Werror=implicit-int]
>   133 | extern FishEncrypt _((char *, char *, char *, int, int));
>       |        ^~~~~~~~~~~
> ```
> 
> 
>      
>   
> 
> Full log: http://sprunge.us/G4LAtj

Probably missed that because of use flags. Patch was so large I didn’t pay attention to them.


Anyway, back to the drawing board

**Comment by Schievel1** [comment]
Should be fine now.

**Comment by thesamesam** [comment]
Lots of these:
```
alias.c:3787:1: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
function_filestat(input)
```

I'm going to pass gnu89 for now.

**Pull request closed without merging**